### PR TITLE
Modernizes Freelancers

### PR DESCRIPTION
**Pull request by hyper2snyper**
## About The Pull Request

Updates the Freelancers equipment, making them more comparable to regular marines.

Freelancer Leader:

Leader

Freelancer Medic:

Medic

Freelancer Standard:

Standard

I have also allowed the armor type /storage to have pockets set to null. This is for ease of modularizing things.
I also fixed a runtime with valk i found.

## Why It's Good For The Game

The ERTs havent been maintained recently, and due to various changes to the marine loadout its been hard to fight xenos as them.
The freelancers didnt even have basic attachments like the magnetic harness. The medic didnt even have tweezers.
This aims to give the Freelancers some standing in the current meta. The freelancer leader will get valk and all the freelancers guns have been given magnetic harnesses.
The freelancers have also been given some optimal storage modules.

This aims to give the freelancers more standing against the xenomorphs, at least a little.
Overall I dont think this will effect their balance at all given the nature of ERTs having to fight in hostile ground mostly. I believe this will serve as a QOL for all those freelancers who lost their gun in the first five  seconds of the conflict.

## Changelog
:cl:
code: Allows the armor type /storage to have pockets set to null.
fix: Fixes a runtime with valk when unequipping it.
balance: Freelancers are modernized!
balance: Freelancer Leader now has a magazine storage module and valk.
He will carry a grenade belt.
His gun now has a mag harness, bayonet and masterkey.
balance: Freelancer Medic now has Baldur, Binocs and a Medical storage module.
He now carries tweezers and MeraDerm.
His Famas now has a bayonet and a magnetic harness.
balance: Freelancer now has baldur and Binocs.
His m16 will now have a magnetic harness, a masterkey and a bayonet.
His stick grenades have been replaced with m15s
He now has a shoulder holster for his sidearm.
He now carries 30 plasteel.
balance: freelancers no longer carry MREs or radios.

/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Neato

**Comment by novaepee** [comment]
Witness the power of mag storage in modular armor.

**Comment by Pill-U** [comment]
Does this change the EORD loadout aswell? Seems a bit strong there, but as EORD is just fun stuff after the game anyways this is just a minor issue.

File: code/datums/components/suit_autodoc.dm [hunk 1]
```diff
@@ -169,7 +169,8 @@
 	SIGNAL_HANDLER
 	if(!iscarbon(user))
 		return
-	UnregisterSignal(wearer, COMSIG_KB_SUITANALYZER)
+	if(wearer)
+		UnregisterSignal(wearer, COMSIG_KB_SUITANALYZER)
```
Comment (TiviPlus) [inline code/datums/components/suit_autodoc.dm:173]: why?

File: code/modules/clothing/modular_armor/attachments/storage.dm [hunk 2]
```diff
@@ -179,6 +188,16 @@
 	icon_state = "mod_medic_bag"
 	storage =  /obj/item/storage/internal/modular/medical
 
+/obj/item/armor_module/storage/medical/freelancer
+
+/obj/item/armor_module/storage/medical/freelancer/Initialize()
```
Comment (TiviPlus) [inline code/modules/clothing/modular_armor/attachments/storage.dm:193]: you dont need an empty type def, byond lets you do it just with redifing a new proc like initialize() here

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/clothing/modular_armor/attachments/storage.dm (hunk 2 above)
Comment (hyper2snyper) [reply code/modules/clothing/modular_armor/attachments/storage.dm:193]: I know but this made me feel better about organization.

File: code/datums/components/suit_autodoc.dm (hunk 1 above)
Comment (hyper2snyper) [reply code/datums/components/suit_autodoc.dm:173]: I mentioned it in the pr description. It was a runtime that was happening with valk and is current. I know it wasn't atomized but it was a single line fix and it was bugging me when testing.

**Comment by Lumipharon** [comment]
Freelancers not having snacks disturbs me. Muh emmulsions.

**Comment by Naaanii** [comment]
Medics finally have ammo.

File: code/modules/clothing/modular_armor/attachments/storage.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/clothing/modular_armor/attachments/storage.dm:193]: it makes it look worse just remove it

**Comment by Lumipharon** [comment]
> Freelancers not having snacks disturbs me. Muh emmulsions.

Also radios are actually useful when comms are down.

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**